src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { RefObject } from 'react';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  newTodoField: RefObject<HTMLInputElement>,
+  setShowError: (val: Errors) => void,
+};
+
+export const NewTodo: React.FC<Props> = ({ newTodoField, setShowError }) => {
+  return (
+    <header className="todoapp__header">
+      <button
+        data-cy="ToggleAllButton"
+        type="button"
+        className="todoapp__toggle-all active"
+      >
+        {}
+      </button>
+
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+          setShowError(Errors.Add);
+        }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)